core/primitives-core/src/parameter.rs

```diff
@@ -0,0 +1,202 @@
+    WasmGrowMemCost,
+    WasmHostFunctionBase,
+    WasmContractLoadingBase,
+    WasmContractLoadingBytes,
+    WasmReadMemoryBase,
+    WasmReadMemoryByte,
+    WasmWriteMemoryBase,
+    WasmWriteMemoryByte,
+    WasmReadRegisterBase,
+    WasmReadRegisterByte,
+    WasmWriteRegisterBase,
+    WasmWriteRegisterByte,
+    WasmUtf8DecodingBase,
+    WasmUtf8DecodingByte,
+    WasmUtf16DecodingBase,
+    WasmUtf16DecodingByte,
+    WasmSha256Base,
+    WasmSha256Byte,
+    WasmKeccak256Base,
+    WasmKeccak256Byte,
+    WasmKeccak512Base,
+    WasmKeccak512Byte,
+    WasmRipemd160Base,
+    WasmRipemd160Block,
+    WasmEcrecoverBase,
+    WasmLogBase,
+    WasmLogByte,
+    WasmStorageWriteBase,
+    WasmStorageWriteKeyByte,
+    WasmStorageWriteValueByte,
+    WasmStorageWriteEvictedByte,
+    WasmStorageReadBase,
+    WasmStorageReadKeyByte,
+    WasmStorageReadValueByte,
+    WasmStorageRemoveBase,
+    WasmStorageRemoveKeyByte,
+    WasmStorageRemoveRetValueByte,
+    WasmStorageHasKeyBase,
+    WasmStorageHasKeyByte,
+    WasmStorageIterCreatePrefixBase,
+    WasmStorageIterCreatePrefixByte,
+    WasmStorageIterCreateRangeBase,
+    WasmStorageIterCreateFromByte,
+    WasmStorageIterCreateToByte,
+    WasmStorageIterNextBase,
+    WasmStorageIterNextKeyByte,
+    WasmStorageIterNextValueByte,
+    WasmTouchingTrieNode,
+    WasmReadCachedTrieNode,
+    WasmPromiseAndBase,
+    WasmPromiseAndPerPromise,
+    WasmPromiseReturn,
+    WasmValidatorStakeBase,
+    WasmValidatorTotalStakeBase,
+    WasmAltBn128G1MultiexpBase,
+    WasmAltBn128G1MultiexpElement,
+    WasmAltBn128PairingCheckBase,
+    WasmAltBn128PairingCheckElement,
+    WasmAltBn128G1SumBase,
+    WasmAltBn128G1SumElement,
+
+    // Smart contract limits
+    MaxGasBurnt,
+    MaxGasBurntView,
+    MaxStackHeight,
+    StackLimiterVersion,
+    InitialMemoryPages,
+    MaxMemoryPages,
+    RegistersMemoryLimit,
+    MaxRegisterSize,
+    MaxNumberRegisters,
+    MaxNumberLogs,
+    MaxTotalLogLength,
+    MaxTotalPrepaidGas,
+    MaxActionsPerReceipt,
+    MaxNumberBytesMethodNames,
+    MaxLengthMethodName,
+    MaxArgumentsLength,
+    MaxLengthReturnedData,
+    MaxContractSize,
+    MaxTransactionSize,
+    MaxLengthStorageKey,
+    MaxLengthStorageValue,
+    MaxPromisesPerFunctionCallAction,
+    MaxNumberInputDataDependencies,
+    MaxFunctionsNumberPerContract,
+    Wasmer2StackLimit,
+    MaxLocalsPerContract,
+}
+
+pub fn load_parameters_from_txt(arg: &[u8]) -> BTreeMap<Parameter, String> {
+    let parameters = BufReader::new(arg).lines().filter_map(|line| {
+        let trimmed = line.unwrap().trim().to_owned();
+        if trimmed.starts_with("#") || trimmed.is_empty() {
+            None
+        } else {
+            let mut iter = trimmed.split(":");
+            let key_str = iter.next().unwrap().trim();
+            let typed_key = Parameter::from_str(key_str)
+                .unwrap_or_else(|_err| panic!("Unexpected parameter `{key_str}`."));
```

Comment:
I would like to see these panics to become regular errors here. There's a lot of value in giving the caller of this function the choice to panic or to handle the error in some other way, I feel. For example it'd be pretty difficult to write unit tests for this functionality (which is something that should be added before landing this PR)

Reply:
That makes sense. What error type would you use here? Something simple as `std::err::Result<_,String>`? Or would you define a new error type in this case?

It's one of those things where I have seen too many solutions over the years but never dug into the details enough to understand which solution is generally accepted to fit which situation.

Reply:
A `thiserror::Error` error type would be most appropriate.

Reply:
Thanks for the advice. Now I propagate a `thiserror::Error` up to `RuntimeConfigStore::new()`, where it still panics. But it is now unit-testable. And when it panics, the error message has context about which file caused the error, so definitely an improvement.